### ISSUE TITLE
Not properly propagating DNS response codes could cause extremely slow DNS queries

### ISSUE THREAD

**Issue by eycorsican** [v2ray/v2ray-core#1565]
It's quite easy to reproduce, just run V2Ray with the following config:
```json
{
    "log": {
        "loglevel": "debug"
    },
    "inbounds": [
        {
            "port": 53,
            "tag": "dns-in",
            "protocol": "dokodemo-door",
            "settings": {
                "address": "1.1.1.1",
                "port": 53,
                "network": "tcp,udp"
            }
        }
    ],
    "dns": {
        "servers": [
            "223.5.5.5"
        ]
    },
    "outbounds": [
        {
            "protocol": "freedom",
            "tag": "direct"
        },
        {
            "protocol": "dns",
            "tag": "dns-out"
        }
    ],
    "routing": {
        "rules": [
            {
                "type": "field",
                "inboundTag": "dns-in",
                "outboundTag": "dns-out"
            },
            {
                "type": "field",
                "ip": [
                    "223.5.5.5"
                ],
                "outboundTag": "direct"
            }
        ],
        "strategy": "rules"
    }
}
```
Issuing the following two commands and you should find they both take a long time to response:
```sh
// Not properly propagates response code NOERROR
dig @127.0.0.1 AAAA www.bilibili.com

// Not properly propagates response code NXDOMAIN
dig @127.0.0.1 A www.bilibili.com.local
```
But without using V2Ray, you should find they return immediately:
```sh
dig @223.5.5.5 AAAA www.bilibili.com
dig @223.5.5.5 A www.bilibili.com.local
```

**Comment by eycorsican** [comment]
I've just tried the fix, it seems still can't handle the case where there are no AAAA records but there is at least one A record. Just the former case above:
```sh
// Not properly propagates response code NOERROR
dig @127.0.0.1 AAAA www.bilibili.com
```

**Comment by VictoriaRaymond** [comment]
Should be fixed by bb8465e1d6e7ef16b86cf0c060266ad037887ef5
